### PR TITLE
[Recoveryservices] Removed VMappContainer check in get protectable item and adding PM reviews

### PR DESCRIPTION
**Pull request by siddharth7**
## Checklist

- [Yes] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [Yes] The title of the PR is clear and informative
- [Yes] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [Yes] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs [hunk 1]
```diff
@@ -80,10 +80,7 @@ public override void ExecuteCmdlet()
                     string containerName = "";
                     Dictionary<UriEnums, string> keyValueDict = HelperUtils.ParseUri(ParentID);
                     containerName = HelperUtils.GetContainerUri(keyValueDict, ParentID);
-                    if (containerName.Split(new string[] { ";" }, System.StringSplitOptions.None)[0].ToLower() == "vmappcontainer")
-                    {
-                        backupManagementType = ServiceClientModel.BackupManagementType.AzureWorkload;
-                    }
+                    backupManagementType = ServiceClientModel.BackupManagementType.AzureWorkload;
```
Comment (cormacpayne) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs]: @siddharth7 why was this change made? Was the command previously filtering out the wrong elements?

Comment (siddharth7) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs]: We saw that we don't need this check. BackupManagementType will always be AzureWorkload for all upcoming ContainerTypes

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs [hunk 2]
```diff
@@ -103,7 +103,14 @@ public override void ExecuteCmdlet()
                     azureWorkloadRecoveryConfig.RecoveryPoint = RecoveryPoint;
                     Dictionary<UriEnums, string> keyValueDict = HelperUtils.ParseUri(RecoveryPoint.Id);
                     string containerUri = HelperUtils.GetContainerUri(keyValueDict, RecoveryPoint.Id);
-                    targetServer = containerUri.Split(new string[] { ";" }, StringSplitOptions.None)[3];
+                    try
+                    {
```
Comment (siddharth7) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs]: Similarly here. We have two types of containers. To accommodate SqlAgContainers too, this change was made.

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs [hunk 3]
```diff
@@ -221,12 +228,19 @@ public string GetTargetPath(string path, string name, int offset)
...
+                ResourceIdentifier resourceIdentifier = new ResourceIdentifier();
+                resourceIdentifier.Subscription = ServiceClientAdapter.SubscriptionId;
+                resourceIdentifier.ResourceGroupName = RecoveryPoint.ContainerName.Split(new string[] { ";" }, StringSplitOptions.None)[1];
+                resourceIdentifier.ResourceType = "/VMAppContainer";
+                resourceIdentifier.ResourceName = RecoveryPoint.ContainerName.Split(new string[] { ";" }, StringSplitOptions.None)[2];
+                return resourceIdentifier.ToString();
+            }
+            catch
+            {
+                return null;
```
Comment (siddharth7) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs]: SQLAgContainers don't require SourceResourceId, hence returning null.

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs [hunk 4]
```diff
@@ -133,11 +149,30 @@ public override void ExecuteCmdlet()
 
                 if (ParameterSetName == FilterParamSet)
                 {
-                    string protectableItemType = ItemType.ToString();
-                    itemModels = itemModels.Where(itemModel =>
+                    if (ItemType != 0)
```
Comment (cormacpayne) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs]: @siddharth7 any reason you're not comparing by enum value, but rather integer value?

Comment (siddharth7) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/GetAzureRMRecoveryServicesBackupProtectableItem.cs]: Nothing specific here, followed the way it was being done in the same file

File: src/RecoveryServices/RecoveryServices/ChangeLog.md [hunk 5]
```diff
@@ -20,6 +20,7 @@
 ## Upcoming Release
 * Added Sql server in Azure VM support
 * SDK Update
+* Removed VMappContainer check in Get-ProtectableItem
```
[the remark this answers is not in the document]
Comment (siddharth7) [reply src/RecoveryServices/RecoveryServices/ChangeLog.md]: Added

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectableItem.md [hunk 6]
```diff
@@ -94,6 +95,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Name
+Name
```
[the remark this answers is not in the document]
Comment (siddharth7) [reply src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupProtectableItem.md]: Updated with texts from PM

**Pull request merged**